src/Sarif/ExtensionMethods.cs

```diff
@@ -15,6 +15,37 @@ namespace Microsoft.CodeAnalysis.Sarif
 {
     public static class ExtensionMethods
     {
+        /// <summary>
+        /// Attempt to reconstruct a URI, if appropriate, using Run instance 
+        /// originalUriBaseId and uriBaseId properties. If this method cannot
+        /// successfully reconstitute an absolute URI, it will return false
+        /// and populate 'resolvedUri' with the input value.
+        /// </summary>
+        /// <param name="run"></param>
+        /// <param name="uri"></param>
+        /// <param name="resolvedUri"></param>
+        /// <returns></returns>
+        public static bool TryResolveUri(this Run run, FileLocation fileLocation, out Uri resolvedUri)
```

Comment:
>Run [](start = 46, length = 3)

You can reduce coupling by just passing the the `originalUriBaseIds` dictionary, rather than the entire `Run` object. That's all you need from the `Run`.

Reply:
I understand that then, what you have is no longer an extension method on `Run`. But -- hah! -- then it could be an extension method on `fileLocation`, which I actually think is more natural:
```
public static bool TryResolveUri(this FileLocation fileLocation, IDictionary<string, string> originalUriBaseIds, out Uri resolvedUri)
```

---
In reply to: [205979411](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205979411) [](ancestors = 205979411)

Reply:
Yes. Absolutely right.

---
In reply to: [205979511](https://github.com/Microsoft/sarif-sdk/pull/974#discussion_r205979511) [](ancestors = 205979511,205979411)